addons/website/static/src/snippets/s_website_form/options.js

```diff
@@ -438,6 +439,48 @@ options.registry.WebsiteFormEditor = FormEditor.extend({
         }
         this._addHiddenField(value, fieldName);
     },
+    /**
+     * Prompts the user to save changes before being redirected
+     * towards an action specified in value.
+     *
+     * @see this.selectClass for parameters
+     */
+    promptSaveRedirect: function (name, value, widgetValue) {
+        return new Promise(resolve => {
+            Dialog.confirm(this, _t(
+                "Would you like to save before being redirected? Unsaved changes will be discarded."
+            ), {
+                cancel_callback: () => resolve(),
+                buttons: [
+                    {
+                        text: _t("Save"),
+                        classes: 'btn-primary',
+                        close: true,
+                        click: () => {
+                            this.trigger_up('request_save', {
+                                reload: false,
+                                onSuccess: () => {
+                                    this._redirectToAction(value);
+                                },
+                            });
+                            resolve();
+                        },
+                    }, {
+                        text: _t("Discard"),
+                        close: true,
+                        click: () => {
+                            this._redirectToAction(value)
+                            resolve();
```

Comment:
@rdeodoo As you want for this one. I think I would not resolve but also not close the dialog (ideally, adding an effect on the button to indicate it is loading).